design/006-proposal-id-generation.md

```diff
@@ -0,0 +1,141 @@
+# Proposal: Mechanism for generating proposal ID
+
+* Author(s): Manoranjith, Matthias Geihs
+* Status: propose
+* Related issue: https://github.com/hyperledger-labs/go-perun/issues/167 
+
+## Summary
+
+This proposal describes the need for proposal ID and the mechanism for
+generating it.
+
+## Motivation
+
+Until now, Proposal ID was being generated by encoding all the parameters of a
+proposal using perunio encoding scheme. Recently, an abstraction for wire
+serialization was introduced (see
+[#233](https://github.com/hyperledger-labs/go-perun/issues/233)), so that
+different encoding schemes can be used for serializing wire messages. As a part
+of this abstraction, it was needed to the mechanism for generating proposal ID
+re-evaluate, because it depends on a specific encoding scheme: perunio
```

Comment:
```suggestion
of this abstraction, we needed to re-evaluate the mechanism for generating the proposal ID, because it depended on a specific encoding scheme: perunio
```

Reply:
Or in passive voice:

"of this abstraction, it was needed to re-evaluate the mechanism for generating the proposal ID, because it depended on a specific encoding scheme: perunio"

Reply:
Updated in 18b56f0b.